Scripts/Services/Pathing/FastAStarAlgorithm.cs

```diff
@@ -42,20 +42,20 @@ public int Heuristic(int x, int y, int z)
 
         public override bool CheckCondition(IPoint3D p, Map map, Point3D start, Point3D goal)
         {
-            return Utility.InRange(start, goal, AreaSize);
+            return Utility.InRange(start, goal, AreaSize - 1);
         }
 
         public override Direction[] Find(IPoint3D p, Map map, Point3D start, Point3D goal)
         {
-            if (!Utility.InRange(start, goal, AreaSize))
+            if (!Utility.InRange(start, goal, AreaSize - 1))
                 return null;
 
             m_Touched.SetAll(false);
 
             m_Goal = goal;
 
-            m_xOffset = (start.X + goal.X - AreaSize) / 2;
-            m_yOffset = (start.Y + goal.Y - AreaSize) / 2;
+	    m_xOffset = (int)Math.Ceiling( (start.X + goal.X - AreaSize) / 2.0 );
```

Comment:
Can we find a better way to do this without floats and ceiling? 🤔

Reply:
From my experiments with this algorithm, the cost of this operation is negligible (it even falls outside of the outermost loop). But one possible alternative is to use `Utility.InRange(start, goal, AreaSize - 2)` instead of `Utility.InRange(start, goal, AreaSize - 1)` as suggested above. Then the problematic edge case is avoided.